server/routes/createPlan/checkYourAnswers/checkYourAnswersController.ts

```diff
@@ -45,133 +45,157 @@ export default class CheckYourAnswersController {
-          modifiedBy: undefined,
-          modifiedDateTime: undefined,
-        },
-        inPrisonInterests:
-          record.inPrisonWork && record.inPrisonEducation
-            ? {
-                inPrisonWork: record.inPrisonWork,
-                inPrisonWorkOther: record.inPrisonWorkOther,
-                inPrisonEducation: record.inPrisonEducation,
-                inPrisonEducationOther: record.inPrisonEducationOther,
-                modifiedBy: undefined,
-                modifiedDateTime: undefined,
-              }
-            : undefined,
-        prisonId: prisoner.prisonId,
-        prisonName: prisoner.prisonName,
-        createdBy: undefined,
-        createdDateTime: undefined,
-        modifiedBy: undefined,
-        modifiedDateTime: undefined,
+      // Check if the Induction is being updated
+      if (getSessionData(req, ['isUpdateFlow', id])) {
+        await this.updateInduction(req, res)
+        res.redirect(addressLookup.learningPlan.profile(id))
+      } else {
+        // else we are creating a new Induction
+        await this.createNewInduction(req, res)
+        res.redirect(`${config.learningPlanUrl}/plan/${id}/induction-created`)
       }
+    } catch (err) {
+      next(err)
+    }
+  }
 
-      // TODO - RR-527 - use CiagPlan instead
-      // Setup required data for api
-      const newRecord = {
-        prisonerId: id,
-        bookingId: prisoner.bookingId,
-        currentUser: res.locals.user.username,
-        hopingToGetWork: record.hopingToGetWork,
-        reasonToNotGetWork: record.reasonToNotGetWork,
-        reasonToNotGetWorkOther: record.reasonToNotGetWorkOther,
-        abilityToWork: record.abilityToWork,
-        abilityToWorkOther: record.abilityToWorkOther,
-        hasWorkedBefore: record.hasWorkedBefore,
-        typeOfWorkExperience: record.typeOfWorkExperience,
-        typeOfWorkExperienceOther: record.typeOfWorkExperienceOther,
-        workExperience: record.workExperience,
-        workInterests: record.workInterests,
-        workInterestsOther: record.workInterestsOther,
-        particularJobInterests: record.particularJobInterests,
-        skills: record.skills,
-        skillsOther: record.skillsOther,
-        personalInterests: record.personalInterests,
-        personalInterestsOther: record.personalInterestsOther,
-        educationLevel: record.educationLevel,
-        qualifications: record.qualifications,
-        additionalTraining: record.additionalTraining,
-        additionalTrainingOther: record.additionalTrainingOther,
-        inPrisonWork: record.inPrisonWork,
-        inPrisonWorkOther: record.inPrisonWorkOther,
-        inPrisonEducation: record.inPrisonEducation,
-        inPrisonEducationOther: record.inPrisonEducationOther,
-        prisonId: prisoner.prisonId,
-        prisonName: prisoner.prisonName,
-      }
+  private async updateInduction(req: Request, res: Response) {
+    const { id } = req.params
+    const { plan } = req.context
```

Comment:
If you want to call this variable `ciagPlan` (or something else?), this is how you would do it:
```
const ciagPlan = req.context.plan
```
(I know I've delayed telling you how to do it - it's easier to explain like that, and also easier to explain when the existing code is only destructuring one object from the request context object)(